src/Http/Event/ImportEventRequestHandler.php

```diff
@@ -52,37 +99,173 @@ public function handle(ServerRequestInterface $request): ResponseInterface
         $responseStatus = $routeParameters->hasEventId() ? StatusCodeInterface::STATUS_OK : StatusCodeInterface::STATUS_CREATED;
+        $eventExists = false;
 
-        /** @var array $data */
-        $data = RequestBodyParserFactory::createBaseParser(
+        if ($routeParameters->hasEventId()) {
+            try {
+                $this->aggregateRepository->load($eventId);
+                $eventExists = true;
+            } catch (AggregateNotFoundException $e) {
+            }
+        }
+
+        /** @var Event $event */
+        $event = RequestBodyParserFactory::createBaseParser(
             $this->combinedRequestBodyParser,
             new IdPropertyPolyfillRequestBodyParser($this->eventIriGenerator, $eventId),
             new JsonSchemaValidatingRequestBodyParser(JsonSchemaLocator::EVENT),
             new CalendarValidationRequestBodyParser(),
             new BookingInfoValidationRequestBodyParser(),
             MainLanguageValidatingRequestBodyParser::createForPlace(),
-            new AssociativeArrayRequestBodyParser()
+            new DenormalizingRequestBodyParser($this->eventDenormalizer, Event::class)
         )->parse($request)->getParsedBody();
 
-        $document = new DecodedDocument($eventId, $data);
+        $eventAdapter = new Udb3ModelToLegacyEventAdapter($event);
+
+        $mainLanguage = $eventAdapter->getMainLanguage();
+        $title = $eventAdapter->getTitle();
+        $type = $eventAdapter->getType();
+        $location = $eventAdapter->getLocation();
+        $calendar = $eventAdapter->getCalendar();
+        $theme = $eventAdapter->getTheme();
+        $publishDate = $eventAdapter->getAvailableFrom(new DateTimeImmutable());
+
+        // Get the workflowStatus from the JSON. If the JSON has no workflowStatus, it will be DRAFT by default.
+        // If the request URL contains "imports", overwrite the workflowStatus to READY_FOR_VALIDATION to ensure
+        // backward compatibility with existing integrations that use those deprecated imports paths without a
+        // workflowStatus, and who expect the workflowStatus to automatically be READY_FOR_VALIDATION or APPROVED.
+        $workflowStatus = $event->getWorkflowStatus();
+        if (str_contains($request->getUri()->getPath(), 'imports')) {
+            $workflowStatus = WorkflowStatus::READY_FOR_VALIDATION();
+        }
+
+        $commands = [];
+        if (!$eventExists) {
+            $eventAggregate = EventAggregate::create(
+                $eventId,
+                $mainLanguage,
+                $title,
+                $type,
+                $location,
+                $calendar,
+                $theme,
+                $publishDate
+            );
+
+            if ($workflowStatus->sameAs(WorkflowStatus::READY_FOR_VALIDATION())) {
+                $eventAggregate->publish($publishDate);
+            }
+
+            // Events created by specific API partners should automatically be approved.
```

Comment:
This is not needed here anymore but should be handled already by https://github.com/cultuurnet/udb3-silex/blob/master/src/Offer/ProcessManagers/AutoApproveForUiTIDv1ApiKeysProcessManager.php

Reply:
It seems like I forgot to remove it from the `ImportPlaceRequestHandler` when I worked on https://github.com/cultuurnet/udb3-silex/pull/946 though. So it should also be removed there. And hopefully the acceptance tests for the auto-approval will still pass then 😅 I'd suggest to remove it from both request handlers and see if the tests are still green. If so it's a small change that can be committed in this PR (just removing this `if` statement below). If it requires additional changes because it ends up not working, we can do it in a separate PR.

Reply:
As discussed this will be done in a new PR because events don't have an acceptance test for this, see ticket https://jira.uitdatabank.be/browse/III-4636